FirebaseRemoteConfig/Sources/FIRRemoteConfig.m

```diff
@@ -700,6 +700,8 @@ - (FIRRemoteConfigSettings *)configSettings {
   dispatch_sync(_queue, ^{
     minimumFetchInterval = self->_settings.minimumFetchInterval;
     fetchTimeout = self->_settings.fetchTimeout;
+    // The NSURLSession needs to be recreated whenever the fetch timeout may be updated.
+    [_configFetch recreateNetworkSession];
```

Comment:
See this is called in only one other place, in below `setConfigSettings` method. My hypothesis is that both/either are called concurrently. 

`[_configFetch recreateNetworkSession]` is not thread-safe so the following happens:

```objc
/// Force a new NSURLSession creation for updated config.
/// - Warning: This API is **not** thread-safe.
- (void)recreateNetworkSession {
  if (_fetchSession) {
    [_fetchSession invalidateAndCancel]; <--- thread A gets here first and runs this (stack trace reveals some async happening in Apple's code)
  }
  _fetchSession = [self newFetchSession]; <--- thread B creates a new URLSession creating a dangling reference for thread A
}
```